lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -435,44 +439,48 @@ def _call(self, method: Callable, params: ApiParams):
                 f"Elasticsearch query failure {e.response.reason} ({e.response.status_code})",
             )
         except requests.exceptions.ConnectionError:
-            self.logger.exception(
+            self.logger.error(
                 "{}: connection refused during the Elasticsearch request", klasname
             )
             raise APIAbort(
                 HTTPStatus.BAD_GATEWAY, "Network problem, could not reach Elasticsearch"
             )
         except requests.exceptions.Timeout:
-            self.logger.exception(
+            self.logger.error(
                 "{}: connection timed out during the Elasticsearch request", klasname
             )
             raise APIAbort(
                 HTTPStatus.GATEWAY_TIMEOUT,
                 "Connection timed out, could reach Elasticsearch",
             )
         except requests.exceptions.InvalidURL:
-            self.logger.exception(
+            self.logger.error(
                 "{}: invalid url {} during the Elasticsearch request", klasname, url
             )
-            raise APIAbort(HTTPStatus.INTERNAL_SERVER_ERROR)
+            raise APIAbort(HTTPStatus.INTERNAL_SERVER_ERROR, "Invalid backend URL")
         except Exception as e:
             self.logger.exception(
                 "{}: exception {} occurred during the Elasticsearch request",
                 klasname,
                 type(e).__name__,
             )
-            raise APIAbort(HTTPStatus.INTERNAL_SERVER_ERROR)
+            raise APIAbort(
+                HTTPStatus.INTERNAL_SERVER_ERROR,
+                f"Unexpected backend exception {str(e)!r}",
+            )
 
         try:
             # postprocess Elasticsearch response
             return self.postprocess(json_response, context)
         except PostprocessError as e:
-            msg = f"{klasname}: the query postprocessor was unable to complete: {e}"
-            self.logger.warning("{}", msg)
+            msg = f"{klasname}: the query postprocessor was unable to complete: {str(e)!r}"
+            self.logger.error("{}", msg)
             raise APIAbort(e.status, str(e.message))
```

Comment:
Why do we report `e.message` in the HTTP response but `str(e)` in the log?  It seems like we would want one or the other and that reporting out something different from what we will look for in the log will be confusing (for us).

Reply:
Yeah, I followed the `{str(e)!r}` pattern despite the `e.message`... the `e` was already here. In fact, for an `e.message` of `f"Can't find Elasticsearch match data {e} in {es_json!r}"`, `str(e)` would be `f"Postprocessing error returning {self.status}: {self.message!r} [{self.data}]"`

Contrast this with the `f"{klasname}: the query postprocessor was unable to complete: {str(e)!r}"` here for `msg`, and it looks like `e.message` (only the root error without context) is too little while the `msg` string using `str(e)` *plus* redundant context is too much. Ugh. Why do you have to see things like this? It just can't be unseen, and now I can't even complain that it's outside the scope of the PR. 😾